eureka/S5_lightcurve_fitting/models.py

```diff
@@ -292,28 +304,46 @@ def _parse_coeffs(self, **kwargs):
         """
 
         # Parse 'c#' keyword arguments as coefficients
-        coeffs = np.zeros(10)
+        coeffs = np.zeros((self.nchan,9))
         for k, v in self.parameters.dict.items():
+            remvisnum=k.split('_')
             if k.lower().startswith('c') and k[1:].isdigit():
-                coeffs[int(k[1:])] = v[0]
+                coeffs[0,int(k[1:])] = v[0]
+            elif len(remvisnum)>1:
+                if remvisnum[0].lower().startswith('c') and remvisnum[0][1:].isdigit() and remvisnum[1].isdigit():
+                    coeffs[int(remvisnum[1]),int(remvisnum[0][1:])] = v[0]
 
         # Trim zeros and reverse
-        self.coeffs = np.trim_zeros(coeffs)[::-1]
+        coeffs=coeffs[:,~np.all(coeffs==0,axis=0)]
+        coeffs=np.flip(coeffs,axis=1)
+        self.coeffs=coeffs
+        # self.coeffs = np.trim_zeros(coeffs)[::-1]
 
     def eval(self, **kwargs):
         """Evaluate the function with the given values"""
         # Get the time
         if self.time is None:
             self.time = kwargs.get('time')
 
-        # Create the polynomial from the coeffs
-        poly = np.poly1d(self.coeffs)
+        longparamlist=self.longparamlist
+        nchan=self.nchan
+        paramtitles=longparamlist[0]
 
         # Convert to local time
         time_local = self.time - self.time.mean()
 
-        # Evaluate the polynomial
-        return np.polyval(poly, time_local)
+        # Create the polynomial from the coeffs
```

Comment:
Can this if/else also be collapsed in a simple way so that unshared fits are just a special case of shared fits? Like what if you set nchan=1 for unshared fits or something like that which would make all the unshared fits work like shared fits without any if/else statements? The only issue I see with that is that some string formatting I wrote uses nchan to figure out how many digits there should be (aka, should if be figure 511 if there's only 9 channels or 5101 if there's more than 10 channels). That should be easy to change though

Reply:
I didn't check the string formatting for the titles, but otherwise this is condensed now.